packages/metadata/test/unit/metadata-accessor.test.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/metadata
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  MetadataAccessor,
+  ClassDecoratorFactory,
+  MetadataInspector,
+} from '../..';
+
+describe('MetadataAccessor', () => {
+  it('creates an accessor with a string key', () => {
+    expect(
+      MetadataAccessor.create<string, ClassDecorator>('foo'),
+    ).to.have.property('key', 'foo');
+  });
+
+  it('overrides toString()', () => {
+    expect(
+      MetadataAccessor.create<object, ClassDecorator>('bar').toString(),
+    ).to.equal('bar');
+  });
+
+  it('can be used to create decorator', () => {
```

Comment:
Might be nitpicky, but I think this test should live in a separate `integration` folder